src/libraries/System.Private.CoreLib/src/System/TimeOnly.cs

```diff
@@ -0,0 +1,905 @@
+
+        /// <summary>
+        /// Determines whether two specified instances of TimeOnly are not equal.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left and right do not represent the same time; otherwise, false.</returns>
+        public static bool operator !=(TimeOnly left, TimeOnly right) => left._ticks != right._ticks;
+
+        /// <summary>
+        /// Determines whether one specified TimeOnly is later than another specified TimeOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is later than right; otherwise, false.</returns>
+        public static bool operator >(TimeOnly left, TimeOnly right) => left._ticks > right._ticks;
+
+        /// <summary>
+        /// Determines whether one specified TimeOnly represents a time that is the same as or later than another specified TimeOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is the same as or later than right; otherwise, false.</returns>
+        public static bool operator >=(TimeOnly left, TimeOnly right) => left._ticks >= right._ticks;
+
+        /// <summary>
+        /// Determines whether one specified TimeOnly is earlier than another specified TimeOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is earlier than right; otherwise, false.</returns>
+        public static bool operator <(TimeOnly left, TimeOnly right) => left._ticks < right._ticks;
+
+        /// <summary>
+        /// Determines whether one specified TimeOnly represents a time that is the same as or earlier than another specified TimeOnly.
+        /// </summary>
+        /// <param name="left">The first object to compare.</param>
+        /// <param name="right">The second object to compare.</param>
+        /// <returns>true if left is the same as or earlier than right; otherwise, false.</returns>
+        public static bool operator <=(TimeOnly left, TimeOnly right) => left._ticks <= right._ticks;
+
+        /// <summary>
+        /// Subtracts a specified time from another specified time and returns a time interval.
+        /// </summary>
+        /// <param name="t1">The time value to subtract from (the minuend).</param>
+        /// <param name="t2">The time value to subtract (the subtrahend).</param>
+        /// <returns>The absolute time interval between t1 and t2. </returns>
+        public static TimeSpan operator -(TimeOnly t1, TimeOnly t2) => new TimeSpan(Math.Abs(t1._ticks - t2._ticks));
```

Comment:
Why do we always want to return a positive value here? I tried javascript and it returns a negative timespan in seconds.

Reply:
The actual use of this API is to calculate the time between 2 times regardless which one before the second. That will allow users to avoid comparing the time before calculating the difference. Users can compare the times if they need to know which comes first.

Reply:
> what this is trying to achieve would be better described as a Between API

That is not true, IsBetween don't return the time between 2 times. If the user wants the time difference, IsBetween cannot help. 

Think about it as time cannot be negative when calculating time intervals. 

@mattjohnsonpint do you have any opinion with this one?

Reply:
It should be positive, yes.  But the implementation needs to change:

```suggestion
        public static TimeSpan operator -(TimeOnly t1, TimeOnly t2) => TimeSpan.FromTicks((t2._ticks - t1._ticks + TimeSpan.TicksPerDay) % TimeSpan.TicksPerDay);
```

Reply:
To expand on *why*, it's because this should give the elapsed time between two points on a circular clock, which will always be a positive value.  (`01:00` - `23:00` = 2 hours)

Reply:
> That is not true, IsBetween don't return the time between 2 times. If the user wants the time difference, IsBetween cannot help.

I was not referring to the `IsBetween` API that we have here, that one makes sense, I was referring to a one that would be named `Between`... anyway sounds good. Just wanted to get clarification on why we needed it to be positive always and makes sense given this is `TimeOnly` and it is referring to a circular clock. Thanks for elaborating.